lib/beaker/hypervisor/aws_sdk.rb

```diff
@@ -69,26 +66,30 @@ def provision
       nil #void
     end
 
-    # Cleanup all earlier provisioned hosts on EC2 using the AWS::EC2 library
-    #
-    # It goes without saying, but a #cleanup does nothing without a #provision
-    # method call first.
+    # Kill all instances.
     #
+    # @param [Enumerable<EC2::Instance>]
     # @return [void]
-    def cleanup
-      @logger.notify("aws-sdk: Cleanup, iterating across all hosts and terminating them")
-      @hosts.each do |host|
-        # This was set previously during provisioning
-        instance = host['instance']
-
-        # Only attempt a terminate if the instance actually is set by provision
-        # and the instance actually 'exists'.
-        if !instance.nil? and instance.exists?
+    def kill_instances(instances)
+      instances.each do |instance|
+        if instance.exists?
+          @logger.notify("aws-sdk: killing EC2 instance #{instance.id}")
           instance.terminate
         end
       end
+      nil
+    end
 
-      nil #void
+    # Cleanup all earlier provisioned hosts on EC2 using the AWS::EC2 library
+    #
+    # It goes without saying, but a #cleanup does nothing without a #provision
+    # method call first.
+    #
+    # @return [unspecified]
```

Comment:
[void] is good enough, quote from yarddoc: http://www.rubydoc.info/gems/yard/file/docs/Tags.md#Literals